extensions/amp-a4a/0.1/test/test-amp-a4a.js

```diff
@@ -128,17 +129,24 @@ describe('amp-a4a', () => {
     resetScheduledElementForTesting(window, 'amp-a4a');
   });
 
-  function createA4aElement(doc) {
+  function createA4aElement(doc, opt_height) {
+    if (opt_height == undefined) {
+      opt_height = 50;
+    }
     const element = createElementWithAttributes(doc, 'amp-a4a', {
       'width': '200',
-      'height': '50',
+      'height': String(opt_height),
       'type': 'adsense',
     });
     element.getAmpDoc = () => {
       const ampdocService = ampdocServiceFor(doc.defaultView);
       return ampdocService.getAmpDoc(element);
     };
     element.isBuilt = () => {return true;};
+    element.getLayoutBox = () => {
+      const visible = element.style.display != 'none';
+      return layoutRectLtwh(0, 0, visible ? 200 : 0, visible ? opt_height : 0);
```

Comment:
But you still have the same issue here, don't you?  If the 'does not initialize promise chain if display none' test passes, you're not sure whether it's because you are detecting `display:none` or because you're detecting the `width == 0`.  Would it work to just get rid of this override of `getLayoutBox` and test height, width, and `display:none` directly, in independent tests?